champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/passives/BloodCompass.java

```diff
@@ -0,0 +1,320 @@
+package me.mykindos.betterpvp.champions.champions.skills.skills.warlock.passives;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import me.mykindos.betterpvp.champions.Champions;
+import me.mykindos.betterpvp.champions.champions.ChampionsManager;
+import me.mykindos.betterpvp.champions.champions.skills.Skill;
+import me.mykindos.betterpvp.champions.champions.skills.types.CooldownSkill;
+import me.mykindos.betterpvp.champions.champions.skills.types.ToggleSkill;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilBlock;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilPlayer;
+import org.bukkit.*;
+import org.bukkit.entity.Player;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.util.Vector;
+
+import java.util.*;
+
+@Singleton
+@BPvPListener
+public class BloodCompass extends Skill implements ToggleSkill, CooldownSkill {
+
+    private int numLines;
+    public double maxDistance;
+    public int effectDuration;
+    public int numPoints;
+    public double escapeRadius;
+    public double damage;
+    public double damageIncreasePerLevel;
+    private final List<List<Location>> markers = new ArrayList<>();
+    private final Map<Integer, Player> lineToPlayerMap = new HashMap<>();
+    private Map<Integer, Integer> currentPoints;
+    private int taskId = -1;
+
+
+
+    @Inject
+    public BloodCompass(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+    @Override
+    public String getName() {
+        return "Blood Compass";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[]{
+                "Drop Sword / Axe to activate",
+                "",
+                "Shoot out up to <val>" + getFinalNumLines(level) + "</val> tracking lines that will fly",
+                "towards the nearest enemies within <stat>" + maxDistance + "</stat> blocks",
+                "",
+                "Players hit with these blood lines will receive",
+                "<effect>Glowing</effect> for <stat>" + effectDuration + "</stat> seconds and take <val>" + getFinalDamage(level) +"</val> damage",
+                "",
+                "Cooldown: <val>" + getCooldown(level)
+        };
+    }
+
+    public double getFinalDamage(int level){
+        return (damage + ((level - 1)) * damageIncreasePerLevel);
+    }
+
+    public int getFinalNumLines(int level){
+        return (numLines + (level - 1));
```

Comment:
It would be nice to have an IncreasePerLevel here, though that might just be me being pedantic.